<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Jianfeng Li" />

<meta name="date" content="2017-06-24" />

<title>Introduction to BioInstaller</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Introduction to BioInstaller</h1>
<h4 class="author"><em>Jianfeng Li</em></h4>
<h4 class="date"><em>2017-06-24</em></h4>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p><a href="https://conda.io/docs/intro.html">Conda</a> and <a href="https://biocontainers.pro">BioContainer</a> have made it easy to install many packages and bio-softwares conveniently. Yet, learning how to install and compile bioinformatics softwares were still necessary. Because, the experience will help you to improve the ability of debugging.</p>
<p>Especialy, when start a NGS analysis work in a new computer or system, you need costs so much time and energy to establish a complete set of softwares and dependce of a analysis pipeline and set the corresponding configuration file.</p>
<p><a href="https://github.com/JhuangLab/BioInstaller">BioInstaller</a> can be used to download/install bioinformatics tools, dependences and databases in R relatively easily, and the information of installed softwares will be saved which can be used to generate configuration file.</p>
<p>Moreover, BioInstaller provide a different way to provide softwares download/install for others.</p>
<p><strong>Feature</strong>:</p>
<ul>
<li>Extendible</li>
<li>Craw the source code and version information from the original site</li>
<li>One step installation or download softwares and databases (Partial dependence supported)</li>
</ul>
</div>
<div id="core-function-in-bioinstaller" class="section level2">
<h2>Core function in BioInstaller</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(BioInstaller)

<span class="co"># Show all avaliable softwares/dependece in default inst/extdata/github.toml </span>
<span class="co"># and inst/extdata/nongithub.toml</span>
<span class="kw">install.bioinfo</span>(<span class="dt">show.all.names =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt;   [1] &quot;abyss&quot;                 &quot;asap&quot;                 </span>
<span class="co">#&gt;   [3] &quot;backspin&quot;              &quot;bamtools&quot;             </span>
<span class="co">#&gt;   [5] &quot;bamutil&quot;               &quot;bcftools&quot;             </span>
<span class="co">#&gt;   [7] &quot;bearscc&quot;               &quot;bedtools&quot;             </span>
<span class="co">#&gt;   [9] &quot;bowtie&quot;                &quot;bowtie2&quot;              </span>
<span class="co">#&gt;  [11] &quot;breakdancer&quot;           &quot;brie&quot;                 </span>
<span class="co">#&gt;  [13] &quot;bwa&quot;                   &quot;cnvkit&quot;               </span>
<span class="co">#&gt;  [15] &quot;cnvnator&quot;              &quot;delly&quot;                </span>
<span class="co">#&gt;  [17] &quot;fastx_toolkit&quot;         &quot;freebayes&quot;            </span>
<span class="co">#&gt;  [19] &quot;fsclvm&quot;                &quot;github_demo&quot;          </span>
<span class="co">#&gt;  [21] &quot;hisat2&quot;                &quot;htseq&quot;                </span>
<span class="co">#&gt;  [23] &quot;igraph&quot;                &quot;isop&quot;                 </span>
<span class="co">#&gt;  [25] &quot;jvarkit&quot;               &quot;libgtextutils&quot;        </span>
<span class="co">#&gt;  [27] &quot;lofreq&quot;                &quot;macs&quot;                 </span>
<span class="co">#&gt;  [29] &quot;mdseq&quot;                 &quot;mimosca&quot;              </span>
<span class="co">#&gt;  [31] &quot;oases&quot;                 &quot;outrigger&quot;            </span>
<span class="co">#&gt;  [33] &quot;picard&quot;                &quot;pindel&quot;               </span>
<span class="co">#&gt;  [35] &quot;pxz&quot;                   &quot;raceid&quot;               </span>
<span class="co">#&gt;  [37] &quot;rca&quot;                   &quot;rum&quot;                  </span>
<span class="co">#&gt;  [39] &quot;samtools_old&quot;          &quot;sclvm&quot;                </span>
<span class="co">#&gt;  [41] &quot;scnorm&quot;                &quot;seqtk&quot;                </span>
<span class="co">#&gt;  [43] &quot;seurat&quot;                &quot;singlesplice&quot;         </span>
<span class="co">#&gt;  [45] &quot;sleuth&quot;                &quot;somaticsniper&quot;        </span>
<span class="co">#&gt;  [47] &quot;sparsehash&quot;            &quot;speedseq&quot;             </span>
<span class="co">#&gt;  [49] &quot;star&quot;                  &quot;tmap&quot;                 </span>
<span class="co">#&gt;  [51] &quot;tophat2&quot;               &quot;tracer&quot;               </span>
<span class="co">#&gt;  [53] &quot;trinityrnaseq&quot;         &quot;varscan2&quot;             </span>
<span class="co">#&gt;  [55] &quot;vcflib&quot;                &quot;vcftools&quot;             </span>
<span class="co">#&gt;  [57] &quot;zifa&quot;                  &quot;annovar&quot;              </span>
<span class="co">#&gt;  [59] &quot;armadillo&quot;             &quot;bcl2fastq2&quot;           </span>
<span class="co">#&gt;  [61] &quot;blat&quot;                  &quot;bzip2&quot;                </span>
<span class="co">#&gt;  [63] &quot;cesa&quot;                  &quot;cnvnator_samtools&quot;    </span>
<span class="co">#&gt;  [65] &quot;curl&quot;                  &quot;edena&quot;                </span>
<span class="co">#&gt;  [67] &quot;ensemble_grch37_reffa&quot; &quot;ensemble_grch38_reffa&quot;</span>
<span class="co">#&gt;  [69] &quot;fastqc&quot;                &quot;fatotwobit&quot;           </span>
<span class="co">#&gt;  [71] &quot;fusioncatcher&quot;         &quot;fusioncatcher_reffa&quot;  </span>
<span class="co">#&gt;  [73] &quot;gatk&quot;                  &quot;gatk_bundle&quot;          </span>
<span class="co">#&gt;  [75] &quot;gmap&quot;                  &quot;hisat2_reffa&quot;         </span>
<span class="co">#&gt;  [77] &quot;htslib&quot;                &quot;imagej&quot;               </span>
<span class="co">#&gt;  [79] &quot;liftover&quot;              &quot;lzo&quot;                  </span>
<span class="co">#&gt;  [81] &quot;lzop&quot;                  &quot;mapsplice2&quot;           </span>
<span class="co">#&gt;  [83] &quot;mutect&quot;                &quot;novoalign&quot;            </span>
<span class="co">#&gt;  [85] &quot;pcre&quot;                  &quot;pigz&quot;                 </span>
<span class="co">#&gt;  [87] &quot;prinseq&quot;               &quot;r&quot;                    </span>
<span class="co">#&gt;  [89] &quot;reditools&quot;             &quot;root&quot;                 </span>
<span class="co">#&gt;  [91] &quot;samstat&quot;               &quot;samtools&quot;             </span>
<span class="co">#&gt;  [93] &quot;snpeff&quot;                &quot;solexaqa&quot;             </span>
<span class="co">#&gt;  [95] &quot;sqlite&quot;                &quot;sratools&quot;             </span>
<span class="co">#&gt;  [97] &quot;ssaha2&quot;                &quot;svtoolkit&quot;            </span>
<span class="co">#&gt;  [99] &quot;tvc&quot;                   &quot;ucsc_reffa&quot;           </span>
<span class="co">#&gt; [101] &quot;ucsc_utils&quot;            &quot;velvet&quot;               </span>
<span class="co">#&gt; [103] &quot;xz&quot;                    &quot;zlib&quot;</span>

<span class="co"># Fetching versions of softwares</span>
<span class="kw">install.bioinfo</span>(<span class="st">'samtools'</span>, <span class="dt">show.all.versions =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt; INFO [2017-06-24 00:37:27] Fetching samtools versions....</span>
<span class="co">#&gt;  [1] &quot;1.5&quot;        &quot;1.4.1&quot;      &quot;1.4&quot;        &quot;1.3.1&quot;      &quot;1.3&quot;       </span>
<span class="co">#&gt;  [6] &quot;1.2&quot;        &quot;1.1&quot;        &quot;1.0&quot;        &quot;0.2.0-rc12&quot; &quot;0.2.0-rc11&quot;</span>
<span class="co">#&gt; [11] &quot;0.2.0-rc10&quot; &quot;0.2.0-rc9&quot;  &quot;0.2.0-rc8&quot;  &quot;0.2.0-rc7&quot;  &quot;0.2.0-rc6&quot; </span>
<span class="co">#&gt; [16] &quot;0.2.0-rc5&quot;  &quot;0.2.0-rc4&quot;  &quot;0.2.0-rc3&quot;  &quot;0.2.0-rc2&quot;  &quot;0.2.0-rc1&quot; </span>
<span class="co">#&gt; [21] &quot;0.1.20&quot;     &quot;0.1.19&quot;     &quot;0.1.18&quot;     &quot;0.1.17&quot;     &quot;0.1.16&quot;    </span>
<span class="co">#&gt; [26] &quot;0.1.15&quot;     &quot;0.1.14&quot;     &quot;0.1.13&quot;     &quot;0.1.12&quot;     &quot;master&quot;</span>

<span class="co"># Install 'demo' quite</span>
download.dir &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">'%s/demo_1'</span>, <span class="kw">tempdir</span>())
<span class="kw">install.bioinfo</span>(<span class="st">'demo'</span>, <span class="dt">download.dir =</span> download.dir, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)
<span class="co">#&gt; $fail.list</span>
<span class="co">#&gt; [1] &quot;&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $success.list</span>
<span class="co">#&gt; [1] &quot;demo&quot;</span>

<span class="co"># Install 'demo' with debug infomation</span>
download.dir &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">'%s/demo_2'</span>, <span class="kw">tempdir</span>())
<span class="kw">install.bioinfo</span>(<span class="st">'demo'</span>, <span class="dt">download.dir =</span> download.dir, <span class="dt">verbose =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt; INFO [2017-06-24 00:37:33] Debug:name:demo</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:33] Debug:destdir:</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:33] Debug:db:~/.BioInstaller</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:33] Debug:github.cfg:/tmp/Rtmp4fAUbp/Rinst137eb6da2ff95/BioInstaller/extdata/github.toml</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:33] Debug:nongithub.cfg:/tmp/Rtmp4fAUbp/Rinst137eb6da2ff95/BioInstaller/extdata/nongithub.toml</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:33] Fetching demo versions....</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:34] Install versions:GRCh37</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:34] Now start to install demo in /tmp/RtmpueRscR/demo_2.</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:34] Running before install steps.</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:34] Now start to download demo in /tmp/RtmpueRscR/demo_2.</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:35] Running install steps.</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:35] Running after install successful steps.</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:35] Running CMD:echo 'successful!'</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:35] Running change.info for demo and be saved to ~/.BioInstaller</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:39] Debug:Install by Github configuration file: </span>
<span class="co">#&gt; INFO [2017-06-24 00:37:39] Debug:Install by Non Github configuration file: demo</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:39] Installed successful list: demo</span>
<span class="co">#&gt; $fail.list</span>
<span class="co">#&gt; [1] &quot;&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $success.list</span>
<span class="co">#&gt; [1] &quot;demo&quot;</span>

<span class="co"># Download demo source code</span>
download.dir &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">'%s/demo_3'</span>, <span class="kw">tempdir</span>())
<span class="kw">install.bioinfo</span>(<span class="st">'demo'</span>, <span class="dt">download.dir =</span> download.dir,
  <span class="dt">download.only =</span> <span class="ot">TRUE</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt; INFO [2017-06-24 00:37:39] Debug:name:demo</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:39] Debug:destdir:</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:39] Debug:db:~/.BioInstaller</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:39] Debug:github.cfg:/tmp/Rtmp4fAUbp/Rinst137eb6da2ff95/BioInstaller/extdata/github.toml</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:39] Debug:nongithub.cfg:/tmp/Rtmp4fAUbp/Rinst137eb6da2ff95/BioInstaller/extdata/nongithub.toml</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:40] Fetching demo versions....</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:40] Install versions:GRCh37</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:40] Now start to download demo in /tmp/RtmpueRscR/demo_3.</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:40] demo be downloaded in /tmp/RtmpueRscR/demo_3 successful</span>
<span class="co">#&gt; [1] TRUE</span>

<span class="co"># Set download.dir rrr destdir (destdir like /usr/local </span>
<span class="co"># including bin, lib, include and others), </span>
<span class="co"># destdir will work if install step {{destdir}} be used</span>
download.dir &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">'%s/demo_source'</span>, <span class="kw">tempdir</span>())
destdir &lt;-<span class="st"> </span><span class="kw">sprintf</span>(<span class="st">'%s/demo'</span>, <span class="kw">tempdir</span>())
<span class="kw">install.bioinfo</span>(<span class="st">'demo'</span>, <span class="dt">download.dir =</span> download.dir, <span class="dt">destdir =</span> destdir)
<span class="co">#&gt; INFO [2017-06-24 00:37:40] Debug:name:demo</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:40] Debug:destdir:/tmp/RtmpueRscR/demo</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:40] Debug:db:~/.BioInstaller</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:40] Debug:github.cfg:/tmp/Rtmp4fAUbp/Rinst137eb6da2ff95/BioInstaller/extdata/github.toml</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:40] Debug:nongithub.cfg:/tmp/Rtmp4fAUbp/Rinst137eb6da2ff95/BioInstaller/extdata/nongithub.toml</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:41] Fetching demo versions....</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:41] Install versions:GRCh37</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:41] Now start to install demo in /tmp/RtmpueRscR/demo.</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:41] Running before install steps.</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:41] Now start to download demo in /tmp/RtmpueRscR/demo_source.</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:41] Running install steps.</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:41] Running after install successful steps.</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:41] Running CMD:echo 'successful!'</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:41] Running change.info for demo and be saved to ~/.BioInstaller</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:46] Debug:Install by Github configuration file: </span>
<span class="co">#&gt; INFO [2017-06-24 00:37:46] Debug:Install by Non Github configuration file: demo</span>
<span class="co">#&gt; INFO [2017-06-24 00:37:46] Installed successful list: demo</span>
<span class="co">#&gt; $fail.list</span>
<span class="co">#&gt; [1] &quot;&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $success.list</span>
<span class="co">#&gt; [1] &quot;demo&quot;</span></code></pre></div>
</div>
<div id="storage-meta-information-of-databases-and-softwares" class="section level2">
<h2>Storage meta information of databases and softwares</h2>
<p>When I install and download massive softwares and databases, I facing the problem how to found it. If we not to save the meta information when you download or install these softwares or databases, you would be in really dire straits.</p>
<p>In fact, version, path, source code path and update time will be saved if you using BioInstaller to install some of softwares. Moreover, you can use some of function in BioInstaller to modify the information in <code>BIO_SOFWARES_DB_ACTIVE</code> database, a TOML format file.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">temp.db &lt;-<span class="st"> </span><span class="kw">tempfile</span>()
<span class="kw">set.biosoftwares.db</span>(temp.db)
<span class="kw">is.biosoftwares.db.active</span>(temp.db)
<span class="co">#&gt; [1] TRUE</span>
params &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">name =</span> <span class="st">'demo'</span>, <span class="dt">comments =</span> <span class="st">'This is a demo.'</span>)
<span class="kw">do.call</span>(change.info, params)
<span class="co">#&gt; INFO [2017-06-24 00:37:46] Running change.info for demo and be saved to /tmp/RtmpueRscR/file137fb31374993</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="kw">get.info</span>(<span class="st">'demo'</span>)
<span class="co">#&gt; $installed</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $source.dir</span>
<span class="co">#&gt; [1] &quot;&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $bin_dir</span>
<span class="co">#&gt; [1] &quot;&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $executable_files</span>
<span class="co">#&gt; [1] &quot;&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comments</span>
<span class="co">#&gt; [1] &quot;This is a demo.&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,&quot;config&quot;)</span>
<span class="co">#&gt; [1] &quot;demo&quot;</span>
<span class="co">#&gt; attr(,&quot;configtype&quot;)</span>
<span class="co">#&gt; [1] &quot;yaml&quot;</span>
<span class="co">#&gt; attr(,&quot;file&quot;)</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpueRscR/file137fb31374993&quot;</span>
<span class="kw">del.info</span>(<span class="st">'demo'</span>)
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
